core/services/relay/evm/evm.go

```diff
@@ -75,18 +76,39 @@ func (r *Relayer) NewConfigProvider(args relaytypes.RelayArgs) (relaytypes.Confi
 
 type configWatcher struct {
 	utils.StartStopOnce
+	lggr             logger.Logger
 	contractAddress  common.Address
 	contractABI      abi.ABI
 	offchainDigester types.OffchainConfigDigester
 	configPoller     *ConfigPoller
 	chain            evm.Chain
+	new              bool
+	fromBlock        uint64
+	replayCtx        context.Context
+	replayCancel     context.CancelFunc
+	wg               sync.WaitGroup
 }
 
 func (c *configWatcher) Start(ctx context.Context) error {
+	if c.new {
+		// Only replay if its a brand new job.
```

Comment:
what if there's no fromBlock in config?

Reply:
good catch, needs an if != 0 check, I missed that in porting the logic to core